prog/kubernetes/kubernetes_nodepool_nexus.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class Prog::Kubernetes::KubernetesNodepoolNexus < Prog::Base
+  subject_is :kubernetes_nodepool
+
+  def self.assemble(name:, node_count:, kubernetes_cluster_id:)
+    DB.transaction do
+      unless KubernetesCluster[kubernetes_cluster_id]
+        fail "No existing cluster"
+      end
+
+      kn = KubernetesNodepool.create(
+        name: name,
+        node_count: node_count,
+        kubernetes_cluster_id: kubernetes_cluster_id
+      )
+
+      Strand.create(prog: "Kubernetes::KubernetesNodepoolNexus", label: "start") { _1.id = kn.id }
+    end
+  end
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        hop_destroy
+      end
+    end
+  end
+
+  label def start
+    nap 30 unless kubernetes_nodepool.kubernetes_cluster.strand.label == "wait"
+    register_deadline("wait", 120 * 60)
+    hop_bootstrap_worker_vms
+  end
+
+  label def bootstrap_worker_vms
+    hop_wait if kubernetes_nodepool.vms.count >= kubernetes_nodepool.node_count
+    push Prog::Kubernetes::ProvisionKubernetesNode, {"nodepool_id" => kubernetes_nodepool.id, "subject_id" => kubernetes_nodepool.kubernetes_cluster.id}
+  end
+
+  label def wait
+    nap 30
+  end
+
+  label def destroy
+    kubernetes_nodepool.vms.each(&:incr_destroy)
+    kubernetes_nodepool.vms.each { kubernetes_nodepool.remove_vm(_1) }
```

Comment:
It also adds `remove_all_*` helper
https://github.com/ubicloud/ubicloud/pull/2598#discussion_r1919142278
```suggestion
    kubernetes_nodepool.remove_all_vms
```